app/src/org/commcare/dalvik/activities/EntityMapActivity.java

```diff
@@ -1,288 +1,185 @@
+            }
+            final LatLngBounds bounds = builder.build();
+
+            // Move camera to be include all markers
+            // TODO: does this work for 1 marker?
+            mMap.setOnMapLoadedCallback(new GoogleMap.OnMapLoadedCallback() {
+                @Override
+                public void onMapLoaded() {
+                    mMap.animateCamera(CameraUpdateFactory.newLatLngBounds(bounds, MAP_PADDING));
+                }
+            });
+        }
+
+        mMap.setOnInfoWindowClickListener(this);
+
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED
+                || ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
+            mMap.setMyLocationEnabled(true);
+        }
     }
 
     @Override
     protected void onResume() {
         super.onResume();
-        mMyLocationOverlay.enableMyLocation();
-        mMyLocationOverlay.enableCompass();
-    }
 
-    protected boolean isRouteDisplayed() {
-        return false;
-    }
-
-    private Location getLocation() {
-        if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_DENIED ||
-                ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_DENIED) {
-            // TODO PLM: warn user and ask for permissions if the user has disabled them
-            return null;
+        if (mMap != null && (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED
+                || ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_GRANTED)) {
+            mMap.setMyLocationEnabled(true);
         }
-
-        // Get the location manager
-        mLocationManager = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
-        // Define the criteria how to select the locatioin provider -> use
-        // default
-        Criteria criteria = new Criteria();
-        String provider = mLocationManager.getBestProvider(criteria, false);
-        return mLocationManager.getLastKnownLocation(provider);
     }
 
-    private void setLocationBounds(Location location, double[] boundHints) {
-        // Initialize the location fields
-        if (location != null) {
-            double lat = location.getLatitude();
-            double lng = location.getLongitude();
+    @Override
+    protected void onPause() {
+        super.onPause();
 
-            //lLat
-            boundHints[0] = lat -1;
-            //lLng
-            boundHints[1] = lng -1;
+        mMap.setOnMapLoadedCallback(null);  // Avoid memory leak in callback
+        if (mMap != null && (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED

```

Comment:
either remove the null check to `mMap` or wrap ` mMap.setOnMapLoadedCallback(null);` in a null check. Currently `mMap != null` is equiv to a no-op